streams-contrib/streams-persist-elasticsearch/src/main/java/org/apache/streams/elasticsearch/ElasticsearchQuery.java

```diff
@@ -220,15 +203,76 @@ public void calcNext() {
     public void remove() {
     }
 
-    public void cleanUp() {
-    }
-
     protected boolean isCompleted() {
-        return totalRead >= this.limit && hasRecords();
+        return totalRead >= this.limit || !hasRecords();
     }
 
     protected boolean hasRecords() {
-        return scrollPositionInScroll != -1 && (!(this.totalRead > this.limit));
+        return scrollPositionInScroll != -1;
     }
 
-}
+    // copied from elasticsearch
+    // if we need this again we should factor it out into a utility
+    private FilterBuilder andFilters(List<FilterBuilder> filters) {
+        if (filters == null || filters.size() == 0)
+            return null;
+
+        FilterBuilder toReturn = filters.get(0);
+
+        for (int i = 1; i < filters.size(); i++)
+            toReturn = FilterBuilders.andFilter(toReturn, filters.get(i));
+
+        return toReturn;
+    }
+
+    private FilterBuilder orFilters(List<FilterBuilder> filters) {
```

Comment:
You spent more effort typing that comment than it would have taken to fix it. ;)  For consistency, braces should be used everywhere.  We've been bitten by lack of braces already when more code is added and the behavior changes because the maintainer didn't add braces.